### PR TITLE
Redesign hoc input

### LINKED ISSUE #989: Redesign form inputs
https://app.abstract.com/projects/23793d20-e68f-11e8-816f-b177d180e919/branches/a6aed27d-7547-4fdb-8dec-f053bd140871/commits/896d3b42f0a52f59513d9f8e62f0b4b48fbd4c82/files/3157643b-b558-4178-b524-1ae72afecf67/layers/5B5B6448-1BC1-4442-B63E-9C02B9861C2E?collectionId=1f5193f4-95dd-4a1d-b117-59d62d58fc68&collectionLayerId=694d2bb0-4fba-474a-a4f9-cd7766f44ead&mode=build&selected=root-5B5B6448-1BC1-4442-B63E-9C02B9861C2E

### PR DESCRIPTION
**Pull request by alcferreira**
## Closes Issues

* Closes #989

### Reviewer 1

* [ ] Functions properly
* [ ] Good readability, clarity and overall quality
* [ ] Has tests
* [ ] Has docs

### Reviewer 2

* [ ] Functions properly
* [ ] Good readability, clarity and overall quality
* [ ] Has tests
* [ ] Has docs

### CODE REVIEW

File: packages/hoc-input/src/formInput.story.mdx [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+
+A High Order Component that adds behaviour of label support, different styles
+for focus and errors, and validation error messages.
+
+This component uses the html `input` element. So the APIs are very similar.
+
+For example, `value`, `placeholder`, `disabled`, `maxLength`, `onChange` and
+others not listed here, should work as expected, you can assume that the
+property will be forwarded to the core html input element. Ref.:
+[MDN &lt;input&gt;](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input)
```
Comment (vNakamura) [inline packages/hoc-input/src/formInput.story.mdx:21]: This text should be in the `TextInput` docs, no?

The HOC will use whichever component we pass to it.

Comment (alcferreira) [reply packages/hoc-input/src/formInput.story.mdx:21]: Hmmm yeah! It makes sense... I'm gonna move it, thanks!

Comment (alcferreira) [reply packages/hoc-input/src/formInput.story.mdx:21]: Changes addressed 😄

Comment (vNakamura) [reply packages/hoc-input/src/formInput.story.mdx:21]: Cool.
I see the text is duplicated. You moved the text in the readme, but it's still present in this mdx file.

Comment (alcferreira) [reply packages/hoc-input/src/formInput.story.mdx:21]: See again :P

Comment (vNakamura) [reply packages/hoc-input/src/formInput.story.mdx]: lol, you squashed the changes to the wrong commit 😆
But it's fine.

**Pull request merged**